api/clients/buckets/bucket.go

```diff
@@ -334,6 +345,8 @@ func (c Client) list(ctx context.Context) (listResponse, error) {
 	return l, nil
 }
 
+var unmodified = errors.New("unmodified bucket")
```

Comment:
I'm not sure if a sentinel error is the best way to handle this, but I honestly couldn't think of a decent alternative that allows the calling method to discern this case.
It needs to know, as otherwise this case is handled and logged the same as a successful HTTP PUT update.